airflow/plugins/utils.py

```diff
@@ -81,7 +153,369 @@ def get_successfully_downloaded_feeds(execution_date):
+        )[0]
+
+    children = directory.children(fs)
+    if len(children) != 1:
+        raise ValueError(
+            f"found {len(directory.children(fs))} files rather than 1 in the directory {directory.name}"
+        )
+
+    ret = children[0]
+
+    # is there a way to have pydantic check this?
+    if not isinstance(ret, GCSFileInfo):
+        raise ValueError(
+            f"encountered unexpected type {type(ret)} rather than GCSFileInfo"
+        )
+
+    return ret
+
+
+class AirtableGTFSDataExtract(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = prefix_bucket("gs://calitp-airtable")
+    table: ClassVar[str] = "california_transit__gtfs_datasets"
+    partition_names: ClassVar[List[str]] = ["dt", "time"]
+    dt: pendulum.Date
+    time: pendulum.Time
+    records: List[AirtableGTFSDataRecord]
+
+    # TODO: this should probably be abstracted somewhere... it's useful in lots of places, probably
+    @classmethod
+    def get_latest(cls) -> "AirtableGTFSDataExtract":
+        # TODO: this concatenation should live on the abstract base class probably
+        latest = get_latest_file(cls.bucket_table(), keys=cls.partition_names)
+
+        logging.info(
+            f"identified {latest.name} as the most recent extract of gtfs datasets"
+        )
+
+        with get_fs().open(latest.name, "rb") as f:
+            content = gzip.decompress(f.read()).decode()
+
+        records = [
+            AirtableGTFSDataRecord(**json.loads(row)) for row in content.splitlines()
+        ]
+
+        return AirtableGTFSDataExtract(
+            records=records,
+            filename=latest.filename,
+            dt=pendulum.parse(latest.partition["dt"], exact=True),
+            time=pendulum.parse(latest.partition["time"], exact=True),
+        )
+
+
+class GTFSFeedExtractInfo(PartitionedGCSArtifact):
+    # TODO: this should check whether the bucket exists https://stackoverflow.com/a/65628273
+    # TODO: this should be named `gtfs-raw` _or_ we make it dynamic
+    bucket: ClassVar[str] = prefix_bucket("gs://gtfs-schedule-raw")
+    partition_names: ClassVar[List[str]] = ["dt", "base64_url", "time"]
+    config: AirtableGTFSDataRecord
+    response_code: int
+    response_headers: Optional[Dict[str, str]]
+    download_time: pendulum.DateTime
+
+    @property
+    def table(self) -> GTFSFeedType:
+        return self.config.data
+
+    @property
+    def dt(self) -> pendulum.Date:
+        return self.download_time.date()
+
+    @property
+    def base64_url(self) -> str:
+        return self.config.base64_encoded_url
+
+    @property
+    def time(self) -> pendulum.Time:
+        return self.download_time.time()
+
+
+class AirtableGTFSDataRecordProcessingOutcome(ProcessingOutcome):
+    input_type: ClassVar[typing.Type[PartitionedGCSArtifact]] = GTFSFeedExtractInfo
+    extract: Optional[GTFSFeedExtractInfo]
+
+
+# class GTFSFeedExtractOutcome(PartitionedGCSArtifact):
```

Comment:
just to confirm -- I should actually implement this as part of the `gtfs_loader` work, right?

Reply:
Yes I just left this stub in, I can remove it if you'd prefer.